commands/chain.go

```diff
@@ -29,11 +29,20 @@ var chainHeadCmd = &cmds.Command{
 		Tagline: "Get heaviest tipset CIDs",
 	},
 	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
-		headTipset := GetPorcelainAPI(env).ChainHead(req.Context)
-		blocks := headTipset.ToSlice()
-		return re.Emit(blocks)
+		return re.Emit(GetPorcelainAPI(env).ChainHead(req.Context).ToSortedCidSet())
 	},
 	Type: []cid.Cid{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, res []cid.Cid) error {
+			for _, r := range res {
```

Comment:
does this actually work for the sortedcidset? it supports range?

Reply:
^

Reply:
We are ranging over a slice of CID's here though right? I think the cmds library is just casting the sorted set to a slice of CID's.

Reply:
I have manually confirmed `range` works by manually crafting a set of CID's greater than length 1.

Reply:
```
frrist@pine ~/s/g/f/g/commands (feat/fix-chain-head-2392)> go-filecoin chain head --enc=text
zDPWYqFCrMH36tQoVUiLrAaC4GCwK4KefHL9JZXxGkcDZq3gTKLt
zDPWYqFD17Q3JZYcTyCr745zECt872UJueihfeJ4rpTWMYrmkYNU
```
```
frrist@pine ~/s/g/f/g/commands (feat/fix-chain-head-2392)> go-filecoin chain head --enc=json
[{"/":"zDPWYqFCrMH36tQoVUiLrAaC4GCwK4KefHL9JZXxGkcDZq3gTKLt"},{"/":"zDPWYqFD17Q3JZYcTyCr745zECt872UJueihfeJ4rpTWMYrmkYNU"}]
```

Reply:
how does it cast a sortedcidset (a struct) to a slice of cids?